File: src/leetCodeProblems/Strings/LengthOfLastWord58.java
package leetCodeProblems.Strings;

/**
 * LeetCode - https://leetcode.com/problems/length-of-last-word/
 */

public class LengthOfLastWord58 {

    public int lengthOfLastWord(String s) {

        String[] input = s.split(" ");
        return input[input.length-1].length();
    }

    public static void main(String[] args) {
        LengthOfLastWord58 obj = new LengthOfLastWord58();
        System.out.println(obj.lengthOfLastWord("   fly me   to   the moon  "));
    }
}

File: src/leetCodeProblems/BruteForce/LengthOfLastWord58.go
package main

/*
- Leetcode - https://leetcode.com/problems/length-of-last-word
- Time - O(n)
- Space - O(1)
*/

func lengthOfLastWord(s string) int {
	out := 0
	wordLength := 0

	for _, v := range s {
		if string(v) != " " {
			wordLength++
		} else {
			if wordLength != 0 {
				out = wordLength
			}

			wordLength = 0
		}
	}

	if wordLength != 0 {
		out = wordLength
	}
	return out
}

// func main() {
// 	log.Println(lengthOfLastWord("Hello World"))
// 	log.Println(lengthOfLastWord("   fly me   to   the moon  "))
// 	log.Println(lengthOfLastWord("luffy is still joyboy"))
// }
